frontend/src/core/notification/components/NotificationPanel.js

```diff
@@ -0,0 +1,71 @@
+/* @flow */
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+
+import './NotificationPanel.css';
+
+import { actions, NAME } from '..';
+
+import type { NotificationState } from '../reducer';
+
+
+type Props = {|
+    notification: NotificationState,
+|};
+
+type InternalProps = {|
+    ...Props,
+    dispatch: Function,
+|};
+
+
+/**
+ */
+export class NotificationPanelBase extends React.Component<InternalProps> {
+    constructor(props: InternalProps) {
+        super(props);
+
+        this.state = {
+            hiding: false,
+        };
+    }
+
+    componentDidUpdate(prevProps: InternalProps) {
+        if (prevProps.notification.messages[0] !== this.props.notification.messages[0]) {
+            this.setState({ hiding: false });
+            setTimeout(() => {
+                this.setState({ hiding: true });
+            }, 2000);
+        }
+    }
+
+    render() {
+        const { notification } = this.props;
+
+        let hideClass = '';
+        if (notification.messages.length) {
+            hideClass = ' showing';
+        }
+        if (this.state.hiding) {
+            hideClass = ' hide';
+        }
+
+        return <ul className={ 'notification-panel' + hideClass }>
+            { notification.messages.map((notif, index) => {
```

Comment:
This use case in particular (and also some others) makes me wonder whether we should rename:
`code/notification` -> `code/notifications`
`NotificationPanel` -> `NotificationsPanel`

Which would also be consistent with `notifications` in the `UserState` and `UserNotificationsMenu`.

Reply:
The thing is, the UI can only ever show one notification at a time. I've kept that `map` in here because that's how it was implemented in the original code, but if you look at the reducer, the array always has maximum 1 element. So maybe we should just change this code instead? Or are we likely to change this notification system in the future to make it allow chaining notifications or showing multiple at once?

Reply:
Good point. I don't think we'll add ability to display multiple notifications anytime soon.